webapp/channels/src/components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -36,508 +42,296 @@ type Props = PropsFromRedux & {
+        desktop_threads: channelMemberNotifyProps?.desktop_threads || NotificationLevels.ALL,
+        mark_unread: channelMemberNotifyProps?.mark_unread || NotificationLevels.ALL,
+        push,
+        push_threads: channelMemberNotifyProps?.push_threads || NotificationLevels.ALL,
+        ignore_channel_mentions: ignoreChannelMentions,
+        channel_auto_follow_threads: channelMemberNotifyProps?.channel_auto_follow_threads || 'off',
+    };
+}
 
-        return currentUserNotifyProps.push_threads;
-    }
-    return NotificationLevels.ALL;
+type SettingsType = {
+    desktop: ChannelNotifyProps['desktop'];
+    desktop_threads: ChannelNotifyProps['desktop_threads'];
+    mark_unread: ChannelNotifyProps['mark_unread'];
+    push: ChannelNotifyProps['push'];
+    push_threads: ChannelNotifyProps['push_threads'];
+    ignore_channel_mentions: ChannelNotifyProps['ignore_channel_mentions'];
+    channel_auto_follow_threads: ChannelNotifyProps['channel_auto_follow_threads'];
 };
 
-export default class ChannelNotificationsModal extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
+export default function ChannelNotificationsModal(props: Props) {
+    const {formatMessage} = useIntl();
+    const [show, setShow] = useState(true);
+    const [serverError, setServerError] = useState('');
+    const [mobileSettingsSameAsDesktop, setMobileSettingsSameAsDesktop] = useState<boolean>(getUseSameDesktopSetting(props.currentUser.notify_props, props.channelMember?.notify_props));
+    const [settings, setSettings] = useState<SettingsType>(getStateFromNotifyProps(props.currentUser.notify_props, props.channelMember?.notify_props));
 
-        const channelNotifyProps = props.channelMember?.notify_props;
-
-        this.state = {
-            show: true,
-            activeSection: NotificationSections.NONE,
-            serverError: null,
-            ...this.getStateFromNotifyProps(props.currentUser.notify_props, channelNotifyProps),
-        };
+    function handleHide() {
+        setShow(false);
     }
 
-    componentDidUpdate(prevProps: Props) {
-        const prevChannelNotifyProps = prevProps.channelMember && prevProps.channelMember.notify_props;
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props;
-
-        if (!deepEqual(channelNotifyProps, prevChannelNotifyProps)) {
-            this.resetStateFromNotifyProps(this.props.currentUser.notify_props, channelNotifyProps);
+    const handleChange = useCallback((values: Record<string, string>) => {
+        setSettings((prevSettings) => ({...prevSettings, ...values}));
+    }, []);
+
+    const handleMobileSettingsChange = useCallback(() => {
+        setMobileSettingsSameAsDesktop((prevSettings) => !prevSettings);
+        setSettings((prevSettings) => ({...prevSettings, push: prevSettings.desktop, push_threads: prevSettings.desktop_threads}));
+    }, [setSettings, setMobileSettingsSameAsDesktop]);
```

Comment:
I don't think these actually need to depend on setter functions returned by `useState` since I don't think those will ever change